types/types.go

```diff
@@ -390,18 +390,22 @@ func (a *Alert) Merge(o *Alert) *Alert {
 	return &res
 }
 
-// A Muter determines whether a given label set is muted. Implementers that
-// maintain an underlying Marker are expected to update it during a call of
-// Mutes.
+func (a Alert) GetExtraLabels() model.LabelSet {
+	return model.LabelSet{"__generatorURL": model.LabelValue(a.GeneratorURL)}
```

Comment:
__generatorURL is a valid (though odd) label that a Prometheus could send. This is best done via a separate mechanism, rather than trying to make it part of silencing via labels. This should also have no impact on inhibits - the generator URL should never affect alert identity.